src/include/sof/audio/pcm_converter.h

```diff
@@ -91,6 +91,52 @@ pcm_get_conversion_function(enum sof_ipc_frame in,
 	return NULL;
 }
 
+/** \brief PCM conversion functions mapfor different size of valid bit and container. */
+struct pcm_func_vc_map {
+	enum sof_ipc_frame source;	/**< source frame container format */
+	enum sof_ipc_frame valid_src_bits;	/**< source frame format */
+	enum sof_ipc_frame sink;	/**< sink frame container format */
+	enum sof_ipc_frame valid_sink_bits;	/**< sink frame format */
+	pcm_converter_func func; /**< PCM conversion function */
+};
+
+/** \brief Map of formats with dedicated conversion functions. */
+extern const struct pcm_func_vc_map pcm_func_vc_map[];
+
+/** \brief Number of conversion functions. */
+extern const size_t pcm_func_vc_count;
+
+/**
+ * \brief Retrieves PCM conversion function.
+ * \param[in] in Source frame format.
+ * \param[in] in valid Source frame format.
+ * \param[in] out Sink frame format.
+ * \param[in] out valid Sink frame format.
+ */
+static inline pcm_converter_func
+pcm_get_conversion_vc_function(enum sof_ipc_frame in,
+			       enum sof_ipc_frame valid_in_bits,
+			       enum sof_ipc_frame out,
+			       enum sof_ipc_frame valid_out_bits)
+{
+	uint32_t i;
+
+	for (i = 0; i < pcm_func_vc_count; i++) {
+		if (in != pcm_func_vc_map[i].source)
+			continue;
+		if (valid_in_bits != pcm_func_vc_map[i].valid_src_bits)
+			continue;
+		if (out != pcm_func_map[i].sink)
```

Comment:
You are checking against `pcm_func_vc_map` in other cases and here against `pcm_func_map`, is this intentional?

Reply:
thanks,  I will update it.